clientconn.go

```diff
@@ -133,10 +146,13 @@ func (cc *ClientConn) resetTransport(closeTransport bool) error {
 		if closeTransport {
 			t.Close()
 		}
-		newTransport, err := transport.NewClientTransport(cc.dopts.protocol, cc.target, cc.dopts.authOptions)
+		newTransport, err := transport.NewClientTransportTimeout(cc.dopts.protocol, cc.target, cc.dopts.authOptions, cc.dopts.timeout)
```

Comment:
now it seems better if we simply pass cc.dopts toNewClientTransport. You do not need to rename the function.